ui/dashboards/src/context/DashboardProvider/panel-slice.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { PanelDefinition } from '@perses-dev/core';
+import { StateCreator } from 'zustand';
+import { Middleware } from './common';
+
+/**
+ * Slice with the state of Panels, along with any actions that modify only the Panels state.
+ */
+export interface PanelSlice {
```

Comment:
now i'm thinking if we really need this slice... since this is a "base slice", maybe it can just live in DashboardProvider?

Reply:
Yeah, like we talked about offline, the main reason to keep it as a "base slice" is to have imports all go in one direction and not be circular.